docs/docs/developers/versions-updating.md

```diff
@@ -0,0 +1,144 @@
+For Aztec contract files, this should be `aztec-nargo` and for noir-only files this should be `nargo`. Mismatching tools and file types will generate misleading syntax and compiler errors.
+:::caution
+
+This can present confusion when opening older contracts (and dependencies) written in older version of noir, such as:
+- Logs filled with errors from the dependencies
+- Or the LSP fails (re-runs automatically then stops)
+The second point requires a restart of the extension, which you can trigger with the command palette (Ctrl + Shift + P) and typing "Reload Window".
+
+## Updating
+### TL;DR
+
+1. Updating the sandbox and CLI:
+
+```shell
+aztec-up
+```
+
+2. Updating aztec-nr and individual @aztec dependencies:
+
+Inside your project run:
+
+```shell
+cd your/aztec/project
+aztec-cli update . --contract src/contract1 --contract src/contract2
+```
+
+The sandbox must be running for the update command to work. Make sure it is [installed and running](../developers/sandbox/references/sandbox-reference.md).
+
+3. Refer [Migration Notes](../misc/migration_notes.md) on any breaking changes that might affect your dapp
+
+---
+
+There are four components whose versions need to be kept compatible:
+
+1. Aztec Sandbox
+2. Aztec CLI
+3. aztec-nargo
+4. `Aztec.nr`, the Noir framework for writing Aztec contracts
+
+First three are packaged together in docker and are kept compatible by running `aztec-up`.
+But you need to update your Aztec.nr version manually or using `aztec-cli update`.
+
+## Updating Aztec.nr packages
+
+### Automatic update
+
+`aztec-cli` will update your Aztec.nr packages to the appropriate version with the `aztec-cli update` command. Run this command from the root of your project and pass the paths to the folders containing the Nargo.toml files for your projects like so:
+
+```shell
+aztec-cli update . --contract src/contract1 --contract src/contract2
+```
+
+### Manual update
+
+To update the aztec.nr packages manually, update the tags of the `aztec.nr` dependencies in the `Nargo.toml` file.
+
+```diff
+[dependencies]
+-aztec = { git="https://github.com/AztecProtocol/aztec-packages", tag="aztec-packages-v0.7.5", directory="noir-projects/aztec-nr/aztec" }
++aztec = { git="https://github.com/AztecProtocol/aztec-packages", tag="#include_aztec_version", directory="noir-projects/aztec-nr/aztec" }
+-value_note = { git="https://github.com/AztecProtocol/aztec-packages", tag="aztec-packages-v0.7.5", directory="noir-projects/aztec-nr/value-note" }
++value_note = { git="https://github.com/AztecProtocol/aztec-packages", tag="#include_aztec_version", directory="noir-projects/aztec-nr/value-note" }
+```
+
+Go to the contract directory and try compiling it with `aztec-nargo compile` to verify that the update was successful:
+
+```shell
+cd /your/contract/directory
+aztec-nargo compile
+```
+
+If the dependencies fail to resolve ensure that the tag matches a tag in the [aztec-packages repository](https://github.com/AztecProtocol/aztec-packages/tags).
```

Comment:
Again this is unneeded if we use an extremely verbose syntax in the above code block. 

But I would also say, right after that code block, "remember to replace <your-aztec-version-here> with the appropriate [tag](https://github.com/AztecProtocol/aztec-packages/tags), for example `aztec-packages-v0.24.0`"

The idea is to prevent users from failing in the first place, as it makes docs more positive. This paragraph assumes the user has failed and wants to troubleshoot.

Reply:
This was from before. But I think if they are getting an error after doing the updates manually (they can also do with `aztec-cli update`), then verifying against repo tags is an apt way to troubleshoot.